### ISSUE TITLE
Squiggle_generator.sh error

### ISSUE THREAD

**Issue by MaestSi** [arnederoeck/NanoSatellite#8]
Dear NanoSatellite developer,
I am trying out your software, but Squiggle_generator.sh script fails.
It seems that the problem is instruction:
`scrappie squiggle ${prefix}.fasta > ${prefix}.squiggle`
In fact, when I run on my fasta file, this error is printed out:

> /home/simone/MinION/software/NanoSatellite/Squiggle_generator.sh: riga 111: 26897 Not allowed instruction (core dump created) scrappie squiggle ${prefix}.fasta > ${prefix}.squiggle
> Quitting from lines 14-35 (Generated_squiggles_graph_report.Rmd)
> Error in index_split[[i + 1]] : out of limit index
> Calls: <Anonymous> ... withCallingHandlers -> withVisible -> eval -> eval -> read.table
> Execution halted

Any idea about how to solve it?
May it be due to scrappie version? (1.4.0 should be in line with specifications)
Thanks in advance,
Simone

**Comment by arnederoeck** [comment]
Dear Simone,

Thanks for reporting this issue. This is indeed caused by a bug in scrappie.
I got a similar error when I tested scrappie v1.4.1 (installed with anaconda). When I tested the currently latest version (v1.4.2) from the scrappie github page, however, everything worked fine again.
So could you update your scrappie to the latest version?

Best,
Arne

**Comment by MaestSi** [comment]
I currently have issues with the installation of the latest scrappie version; I'll wait until it is going to be available on conda, and I'll try again. Thanks.
Simone

**Comment by arnederoeck** [comment]
Alternatively you could also try an older version of scrappie. 1.3.2 for instance is also available on conda.

**Comment by MaestSi** [comment]
I already tried scrappie 1.3.2, but kept on having the same error.

**Comment by MaestSi** [comment]
Dear Arne,
I installed scrappie v1.4.2 with `conda install -c bioconda scrappie`, as shown by `conda list`.

> scrappie                  1.4.2            py36h0fb1251_0    bioconda

However, `scrappie version` gives 

> scrappie 1.4.0-.

Is this the same version you are using, in your opinion?
Because I still have the same error as with older scrappie versions when I run command  `scrappie squiggle`.
Thanks,
Simone

**Comment by wdecoster** [comment]
I am not sure if that is the expected version for scrappie 1.4.2, but could you make sure that the `scrappie` you are executing is the one installed from conda, and not another installation somewhere on your $PATH?
You can check the path to the executable using `which scrappie`.

Recommended is to create separate conda environments to install tools and keep dependencies separate. In this case, you would do `conda create --name scrappie_env -c bioconda scrappie`

**Comment by MaestSi** [comment]
Sure, I already did that check and already created a dedicated conda environment.
`which scrappie` returns 

> /home/simone/miniconda3/envs/NanoSatellite_env/bin/scrappie

**Comment by arnederoeck** [comment]
Hi,
It seems that the scrappie squiggle github versions are fine, and that the scrappie squiggle conda versions do not work. I've tried all different scrappie versions on conda and different python distributions etc. but unfortunately I can't make it work. So I have contacted the creator and hopefully we can come up with a solution soon.
FYI, I also see the `scrappie 1.4.0-.` output when I check for the version of the latest (1.4.2) conda scrappie version.

Best,
Arne

**Comment by MaestSi** [comment]
Thank you very much. Hope that conda installation of scrappie will be fixed soon!
Best,
Simone

**Comment by arnederoeck** [comment]
Hi,
So, earlier you mentioned that you had problems installing scrappie from github. If this is due to hdf5 dependencies, then you can now try to install the "squiggle_only" branch, which Tim Massingham, the creator of scrappie has now pushed to github (https://github.com/nanoporetech/scrappie/tree/squiggle_only).
This will still require a BLAS library though and I've seen that on some operating systems this forms a problem. I was able to resolve most BLAS related issues by adapting the `CMakeLists.txt` file after cloning the scrappie repository and before building scrappie. So in my case for instance, I changed the following piece of code:
```
##
#   Set up include directories
##
include_directories ("${PROJECT_BINARY_DIR}/include")
if (OPENBLAS_ROOT)
	include_directories ("${OPENBLAS_ROOT}/include")
	link_directories ("${OPENBLAS_ROOT}/lib")
	set (BLAS "openblas")
else ()
	set (BLAS "blas")
endif ()
```
to:

```
##
#   Set up include directories
##
include_directories ("${PROJECT_BINARY_DIR}/include")

include_directories ("/YOUR/PATH/openblas")
link_directories ("${OPENBLAS_ROOT}/lib")
set (BLAS "openblas")
```
In my case `/YOUR/PATH/openblas` corresponds to `/usr/include/openblas`.

Hope this helps!

**Comment by MaestSi** [comment]
Hi,
thank you for trying to fix this. May I ask you how you installed openblas? I tried with `sudo apt-get install libopenblas-dev` but I don't have an openblas directory or header file in /usr/include. I also tried with `conda install -c anaconda openblas` (and apparently seemed to be working too) and with `git clone https://github.com/xianyi/OpenBLAS.git`, but `make` command stopped with errors.
I am working on a Ubuntu 14.04 machine.

**Comment by MaestSi** [comment]
Dear all,
I managed to fix the openblas installation with `pip install configparser`, since that package was missing. Then I compiled squiggle_only branch of scrappie following your suggestions, and finally modified line
`scrappie squiggle ${prefix}.fasta > ${prefix}.squiggle` to:
```
SQUIGGLE=/home/simone/MinION/software/scrappie_squiggle_generator/build/scrappie_squiggle
$SQUIGGLE ${prefix}.fasta > ${prefix}.squiggle
```
 in Squiggle_generator.sh script.
Now, NanoSatellite is working properly, so thank you very much for all your precious advice.
Simone
